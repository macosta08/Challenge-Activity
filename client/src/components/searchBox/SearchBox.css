nav {
  background: #ffe600;
  width: 100%;
  padding: 10px 0px;
  display: inline-flex;
  justify-content: center;
  box-sizing: border-box; /* para ajustarse al tama√±o de la pantalla  */
}

nav .nav-container {
  background: none;
  display: inline-flex;
  width: 90%;
}

nav .nav-container div.logo {
  background-image: url('https://http2.mlstatic.com/frontend-assets/ui-navigation/5.14.3/mercadolibre/logo__small.png');
  background-repeat: no-repeat;
  background-color: #ffe600;
  background-position: center; /* Center the image */
  height: 40px;
  width: 60px;
}

nav .search-box {
  height: 40px;
  width: 100%;
  display: flex;
  margin-left: 20px;
  border-radius: 5px;
}

nav .search-box input {
  height: 100%;
  width: 100%;
  border: none;
  outline: none;
  border-radius: 4px 0 0 4px;
  padding: 0 10px;
  font-size: 16px;
  background: #fff;
}

nav .search-box button {
  height: 100%;
  width: 40px;
  border: none;
  background: #eeeeee;
  border-radius: 0 4px 4px 0;
  outline: none;
  cursor: pointer;
}

input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus,
input:-webkit-autofill:active {
  transition: background-color 5000s;
  -webkit-text-fill-color: black !important;
}

@media (min-width: 640px) {
  nav .nav-container {
    width: 85%;
  }
}

@media (min-width: 768px) {
  nav .nav-container {
    width: 83%;
  }
}

@media (min-width: 1024px) {
  nav .nav-container {
    width: 80%;
  }
  nav .search-box input {
    font-size: 18px;
  }
}

@media (min-width: 1280px) {
  nav .nav-container {
    width: 75%;
  }
}

@media (min-width: 1536px) {
  nav .nav-container {
    width: 70%;
  }
}
