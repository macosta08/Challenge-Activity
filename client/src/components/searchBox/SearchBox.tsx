import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router';
import { useLocation } from 'react-router';
import './SearchBox.css';

const SearchBox = () => {
  const location = useLocation();
  const history = useNavigate();
  const query = location.search.replace('?search=', '').replace(/%20/g, ' ');
  const [stateSearch, setStateSearch] = useState(query);
  const handleInputChange = (e: { target: { value: React.SetStateAction<string> } }) => {
    setStateSearch(e.target.value);
  };

  const handleSearch = (e: { preventDefault: () => void }) => {
    e.preventDefault();
    history(`/items?search=${stateSearch}`);
  };

  useEffect(() => {
    setStateSearch(query);
  }, [query]);

  return (
    <nav>
      <div className='nav-container'>
        <div className='logo' />
        <div className='search-box'>
          <input
            type='text'
            placeholder='Nunca dejes de buscar'
            name='searchText'
            value={stateSearch}
            onChange={handleInputChange}
            onKeyPress={(event) => event.key === 'Enter' && handleSearch(event)}
          />
          <button type='button' onClick={handleSearch}>
            <i className='fa fa-search' />
          </button>
        </div>
      </div>
    </nav>
  );
};

export default SearchBox;
