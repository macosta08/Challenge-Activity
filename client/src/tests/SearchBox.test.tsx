import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import SearchBox from '../components/searchBox/SearchBox';

describe('Prueba al navbar <SearchBox/>', () => {
  const inputValue = 'Laptop';

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(
      <Router>
        <SearchBox />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
  test('Debe hacer event onClick', () => {
    render(
      <Router>
        <SearchBox />
      </Router>
    );
    fireEvent.click(screen.getByRole('button', { name: 'btn-search' }));
  });

  test('debe de llamar handleSearch si el input tiene un valor o al hacer onClick', () => {
    render(
      <Router>
        <SearchBox />
      </Router>
    );
    const handleSearch = jest.fn();
    const input = screen.getByRole('textbox');
    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.click(screen.getByRole('button', { name: 'btn-search' }));

    expect(handleSearch).toHaveBeenCalledTimes(0);
  });
});
