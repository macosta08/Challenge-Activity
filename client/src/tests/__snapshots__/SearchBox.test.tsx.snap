// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba al navbar <SearchBox/> Debe hacer match con el snapshot 1`] = `
<div>
  <nav>
    <div
      class="nav-container"
    >
      <div
        class="logo"
      />
      <div
        class="search-box"
      >
        <input
          name="searchText"
          placeholder="Nunca dejes de buscar"
          type="text"
          value=""
        />
        <button
          aria-label="btn-search"
          type="button"
        >
          <i
            class="fa fa-search"
          />
        </button>
      </div>
    </div>
  </nav>
</div>
`;
